### PR TITLE
add script to generate podspec

### PR DESCRIPTION
**Pull request by tomerd**
motivation: distribute the library via podspec

changes:
* add build_podspec.sh script
* adjust sanity check to support more types of copyright years format

### CODE REVIEW

File: scripts/build_podspec.sh [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  s.version = '$version'
+  s.license = { :type => 'Apache 2.0', :file => 'LICENSE.txt' }
+  s.summary = 'A Metrics API for Swift.'
+  s.homepage = 'https://github.com/apple/swift-metrics'
+  s.author = 'Apple Inc.'
+  s.source = { :git => 'https://github.com/apple/swift-metrics.git', :tag => s.version.to_s }
+  s.documentation_url = 'https://apple.github.io/swift-metrics'
+  s.module_name = '$name'
+
+  s.swift_version = '4.2'
```
Comment (yim-lee) [inline scripts/build_podspec.sh]: Should this section match SwiftNIO's? https://github.com/apple/swift-nio/blob/master/scripts/build_podspecs.sh#L88

Comment (tomerd) [reply scripts/build_podspec.sh]: nio v. 2.0 needs swift 5, this one is okay with > 4.2

File: scripts/build_podspec.sh [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.txt for the list of Swift Metrics API project authors
+##
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+
+##===----------------------------------------------------------------------===##
```
Comment (MrLotU) [inline scripts/build_podspec.sh]: Double header?

File: scripts/sanity.sh [hunk 3]
```diff
@@ -13,9 +13,28 @@
 ##
 ##===----------------------------------------------------------------------===##
 
+##===----------------------------------------------------------------------===##
```
Comment (MrLotU) [inline scripts/sanity.sh]: Double header here too

**Pull request merged**